### PR TITLE
Refactored Challenge Icons to handle icon updates

### PR DESCRIPTION
**Pull request by LillyJadeKatrin**
If an icon is displayed on screen before it downloads, it was displaying a default icon but it would fail to load the actual icon even after it was downloaded. This fixes that.

### CODE REVIEW

File: Source/Core/Core/AchievementManager.cpp [hunk 1]
```diff
@@ -338,7 +338,15 @@ AchievementManager::RichPresence AchievementManager::GetRichPresence() const
   return m_rich_presence;
 }
 
-const AchievementManager::NamedBadgeMap& AchievementManager::GetChallengeIcons() const
+const bool AchievementManager::AreChallengesUpdated()
```
Comment (iwubcode) [inline Source/Core/Core/AchievementManager.cpp:341]: I'm nitpicking here but this feels a bit strange to me, as the name implies a simple `bool` result but it's actually doing some work by resetting the value.

Thoughts on splitting this in two?

```
bool AchievementManager::AreChallengesUpdated() const
{
  return m_challenges_updated;
}

void AchievementManager::ResetChallengesUpdated()
{
  m_challenges_updated = false;
}
```

File: Source/Core/Core/AchievementManager.cpp [hunk 2]
```diff
@@ -796,15 +804,16 @@ void AchievementManager::HandleAchievementChallengeIndicatorShowEvent(
     const rc_client_event_t* client_event)
 {
   auto& instance = AchievementManager::GetInstance();
-  instance.m_active_challenges[client_event->achievement->badge_name] =
-      &AchievementManager::GetInstance().GetAchievementBadge(client_event->achievement->id, false);
+  instance.m_active_challenges.insert(client_event->achievement->id);
```
Comment (iwubcode) [inline Source/Core/Core/AchievementManager.cpp:807]: Should this handle whether the id already existed?

```
const auto [iter, inserted] = instance.m_active_challenges.insert(client_event->achievement->id);
if (inserted)
  instance.m_challenges_updated = true;
```

File: Source/Core/Core/AchievementManager.cpp [hunk 3]
```diff
@@ -796,15 +804,16 @@ void AchievementManager::HandleAchievementChallengeIndicatorShowEvent(
...
+  instance.m_challenges_updated = true;
 }
 
 void AchievementManager::HandleAchievementChallengeIndicatorHideEvent(
     const rc_client_event_t* client_event)
 {
-  AchievementManager::GetInstance().m_active_challenges.erase(
-      client_event->achievement->badge_name);
+  auto& instance = AchievementManager::GetInstance();
+  instance.m_active_challenges.erase(client_event->achievement->id);
```
Comment (iwubcode) [inline Source/Core/Core/AchievementManager.cpp:815]: Should this handle whether the id was actually erased?

```
const auto removed = instance.m_active_challenges.erase(client_event->achievement->id);;
if (removed > 0)
  instance.m_challenges_updated = true;
```

File: Source/Core/Core/AchievementManager.cpp (hunk 2 above)
Comment (LillyJadeKatrin) [reply Source/Core/Core/AchievementManager.cpp:807]: This and the question below are operating under the impression that rc_client knows what it is doing when triggering these events, but I can add a safety check if you'd like.

File: Source/Core/Core/AchievementManager.cpp (hunk 1 above)
Comment (LillyJadeKatrin) [reply Source/Core/Core/AchievementManager.cpp:341]: I don't think I'd split them given that the second method is only ever supposed to be called from the context of the first method. Would CheckForChallengeUpdates be a better name perhaps?

**Pull request merged**